src/main/java/com/celements/web/plugin/cmd/ExternalJavaScriptFilesCommand.java

```diff
@@ -243,9 +253,14 @@ private String addAllExtJSfilesFromDoc(XWikiDocument doc) {
     return jsIncludes2;
   }
 
-  private List<String> getJavaScriptExternalFilePaths(XWikiDocument doc) {
-    List<BaseObject> javaScriptFiles = doc.getXObjects(new DocumentReference(context.getDatabase(),
-        JAVA_SCRIPT_EXTERNAL_FILES_CLASS_SPACE, JAVA_SCRIPT_EXTERNAL_FILES_CLASS_DOC));
+  @NotNull
+  private List<String> getJavaScriptExternalFilePaths(@Nullable XWikiDocument doc) {
+    if (doc == null) {
+      return Collections.emptyList();
+    }
+    List<BaseObject> javaScriptFiles = doc.getXObjects(
+        getOldCoreClasses().getJavaScriptExternalFilesClassRef(
+            doc.getDocumentReference().getWikiReference().getName()));
```

Comment:
Please change jsFiles to ArrayList. There is now need for type Vector in this single threaded situation.